src/pudl_scrapers/bin/ferc_xbrl.py

```diff
@@ -0,0 +1,372 @@
+        if not url.netloc.endswith("ferc.gov"):
+            continue
+
+        # Download file
+        file = requests.get(url.geturl())
+
+        path = Path(url.path).relative_to("/")
+
+        with archive.open(str(path), "w") as f:
+            f.write(file.text.encode("utf-8"))
+
+
+def archive_year(year: Year, filings: set[FeedEntry], form: FercForm, output_dir: Path):
+    """Archive a single year of data for a desired form.
+
+    Args:
+        year: Year to archive.
+        filings: Set of filings indexed from RSS feed.
+        form: Ferc form.
+        output_dir: Directory to save archived filings in.
+    """
+    # Get form number as integer
+    form_number = form.as_int()
+
+    metadata = {}
+    archive_path = output_dir / f"ferc{form_number}-xbrl-{year}.zip"
+
+    with zipfile.ZipFile(archive_path, "w") as archive:
+        # Archive taxonomy
+        # The first version of the taxonomy was released in 2020
+        if year > 2019:
+            archive_taxonomy(form, year, archive)
+
+        for filing in filings:
+            # Download filing
+            xbrl_file = requests.get(filing.download_url)
+
+            # Add filing metadata
+            filing_name = f"{filing.title}{filing.ferc_period}"
+            if filing_name in metadata:
+                metadata[filing_name].update({filing.entry_id: filing.json()})
+            else:
+                metadata[filing_name] = {filing.entry_id: filing.json()}
+
+            # Write to zipfile
+            with archive.open(f"{filing.entry_id}.xbrl", "w") as f:
+                logger.info(f"Writing {filing.title} to form {form_number} archive.")
+                f.write(xbrl_file.text.encode("utf-8"))
+
+        # Save snapshot of RSS feed
+        with archive.open("rssfeed", "w") as f:
+            logger.info("Writing rss feed metadata to archive.")
+            f.write(json.dumps(metadata).encode("utf-8"))
+
+    logging.info(f"Finished scraping ferc{form_number}-{year}.")
+
+
+def archive_form(
+    form: FercForm, indexed_filings: FormFilings, use_latest_dir: bool = False
+):
+    """Create an archive for each year with filings available for the form in question.
+
+    Args:
+        form: Ferc form.
+        indexed_filings: Dictionary mapping available filings to year of filing.
+        use_latest_dir: Use the most recently created output directory.
+    """
+    # Get form number as integer
+    form_number = form.as_int()
+
+    # Create new output directory or use latest
+    if use_latest_dir:
+        output_dir = get_latest_directory(
+            Path(pudl_scrapers.settings.OUTPUT_DIR) / f"ferc{form_number}"
+        )
+    else:
+        # Create output directory if it doesn't exist
+        output_dir = new_output_dir(
+            Path(pudl_scrapers.settings.OUTPUT_DIR) / f"ferc{form_number}"
+        )
+        if not output_dir.exists():
+            output_dir.mkdir(parents=True)
```

Comment:
Could this be simplified to:

```python
output_dir = get_latest_directory(
    Path(pudl_scrapers.settings.OUTPUT_DIR) / f"ferc{form_number}"
)
if not use_latest_dir:
    if not output_dir.exists():
        output_dir.mkdir(parents=True)
```

Reply:
It still needs to create a new output directory if it's not using the latest one. This could all definitely be cleaned up, but I plan on reworking the whole output directory thing when I combine the archiver/scraper, so I haven't done too much just yet

Reply:
Ohhh whoops I thought line 308 and 313 were calling the same function. My bad.